sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java

```diff
@@ -921,15 +1015,19 @@ public void onFailure(@NonNull Call<ResponseBody> call, @NonNull Throwable t) {
                         headers);
 
                     return new CallAndOptionalResult<>(call, result);
-                } else { // Error response
-                    String strContent = readAsString(response.body());
+                } else {
+                    ResponseBody responseBody = response.body();
+                    String message = responseBody == null ? "no message" : "message: " + readAsString(responseBody);
 
-                    throw new BlobStorageException(strContent, response.raw());
+                    throw new BlobStorageException("Response failed with code: " + response.code() + " and " + message,
```

Comment:
Refer above comment i.e. I would suggest keeping the message as it is. Also the HTTP response.code() is often 200 with the actual error-code exists in the response-header, so this extra text additions may not add much value but make it bit difficult to parse like I mentioned above.